CONTRIBUTING.md

```diff
@@ -0,0 +1,17 @@
+# Contributing
+
+## Reporting Issues
+Feel free to open Issues for any Discussions, Bugs or Improvements. Please adhere to the provided Templates and provide the required context. 
+
+## Pull Request Process
+1. Please open an Issue describing the bug or feature you are intending to fix. If you are working on an existing Issue comment on relevant Issue in order to let other Contributors know that you are working on it
+
+2. [Fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo) the Repository using the standard workflow and set up a new branch to work in. Make sure that each group of changes is done in a separate branch, so that PRs only contain relevant changes 
+
+3. This Project uses [RustFmt](https://github.com/rust-lang/rustfmt) to ensure proper formatting. PRs that do not adhere to the formatting will not be able to be merged! The formatting can be done using `cargo fmt --all`. If you are using IntelliJ Idea for developing you can set up RustFmt using the `Languages&Frameworks > Rust > RustFmt` settings
+
+4. This Project uses [Clippy](https://github.com/rust-lang/rust-clippy) as a code linter. Execute it before committing using `cargo clippy -- -D warnings`. Intellij allows you to run Clippy while developing, to find linter errors as they happen
```

Comment:
Actually, the PR verification workflow does more:
```
cargo fmt --all -- --check
cargo check --tests --examples
cargo test --verbose
cargo clippy -- -D warnings
```
And then
```
cargo +nightly clippy --color=always --all-targets -- -W clippy::pedantic
```

Maybe we can simplify this with a shell script that contributors can then use, see #110 (and #109)